### DISCUSSION TITLE
Reading reaction forces from FEMEntity

### DISCUSSION THREAD

**Discussion by bahadir14** [Genesis-Embodied-AI/genesis-world#1679] [Q&A]
Hi, I would like to read reaction forces from free falling cube, which is FEM material. I knew that it exist for rigid entities, but not sure that it exists for FEMEntity. Here is my full code:

*****************************
import argparse

import numpy as np
import os
import torch

from dask.order import order
from mpmath import euler
from pygel3d.hmesh import minimize_dihedral_angle
from os import path
os.environ['PYOPENGL_PLATFORM'] = 'glx'
import genesis as gs

def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("-v", "--vis", action="store_true", default=False)
    args = parser.parse_args()

    ########################## init ##########################
    gs.init(seed=0, precision="32", logging_level="debug")

    ########################## create a scene ##########################

    scene = gs.Scene(
        sim_options=gs.options.SimOptions(
            dt=4e-3,
            substeps=2,
        ),
        #coupler_options=gs.options.CouplerOptions(
        #    rigid_fem=True,
        #),
        rigid_options=gs.options.RigidOptions(
            dt=0.01,
            gravity=(0,0,-9.8),
            enable_collision=True,
            constraint_solver=gs.constraint_solver.Newton,
            iterations=200,
            tolerance=1e-7,
            integrator=gs.integrator.implicitfast,
            #use_contact_island=True,
        ),
        fem_options=gs.options.FEMOptions(
            use_implicit_solver=True,
            damping=0.1,
        ),
        viewer_options=gs.options.ViewerOptions(
            camera_pos=(9.6, 4.8, 14.0),
            camera_lookat=(0.0, 0.0, 0.0),
            camera_fov=35,
            max_FPS=30,
        ),
        show_viewer=True,
        vis_options=gs.options.VisOptions(
            #visualize_mpm_boundary=True,
        ),
    )
    cam = scene.add_camera(
        res=(1280, 800),
        pos=(5.5, 6.5, 3.2),
        lookat=(0.5, 1.5, 1.5),
        fov=30,
        GUI=True,
    )

    plane = scene.add_entity(
        material=gs.materials.Rigid(),
        morph=gs.morphs.URDF(file="urdf/plane/plane.urdf", fixed=True),
    )
    E, nu = 1.e7,0.3
    rho = 2700.0

    obj1 = scene.add_entity(
        material=gs.materials.FEM.Elastic(
            E=E,
            nu=nu,
            rho=rho,
            model="linear",
        ),
        morph=gs.morphs.Box(
            pos=(0.0, 0.0, 5.0),
            size=(0.5,0.5,0.5),
        ),
        surface=gs.surfaces.Metal(
            color=(0.6, 1.0, 0.8, 1.0),
            vis_mode="visual",
        ),
    )

    scene.build()
    cam.start_recording()

    horizon = 1000
    for i in range(horizon):
            #if i < 400:
                #obj1.set_dofs_velocity([0.0, -2.0, 0.0,0.0,0.0,0.0])

            scene.step()
            cam.render()
    cam.stop_recording(save_to_filename="fem_rigid_entities.mp4", fps=60)

if __name__ == "__main__":
    main()
*****************************

I would be happy to see your answers :)
